### DISCUSSION TITLE
How to restore cursor color on exit correctly?

### DISCUSSION THREAD

**Discussion by shmerl** [neovim/neovim#31717] [Q&A] [answered]
I have these settings in my init vimscript configuration which set various blinking and shape parameters and restores cursor on exit, but I noticed that it fails to restore cursor color to terminal's default and cursor color remains as defined by chosen neovim's color theme. It is being restored if I don't use these customizations. Is there some way to do it correctly?

```vimscript
" cursor shapes
set guicursor=n-c:block-Cursor/lCursor-blinkon1,v-i-ci:ver25-Cursor/lCursor-blinkon1,r-cr:hor20-Cursor/lCursor-blinkon1

" restoring cursor to non blinking state on exit
au VimLeave * set guicursor=a:block-blinkon0
```

**Comment by shmerl** [comment]
Any tips?

**Comment by zeertzjq** [comment, marked as answer]
Konsole supports OSC 12 to set cursor color, but not OSC 112 to reset cursor color. You need to request OSC 112 support at <https://invent.kde.org/utilities/konsole>.

**Reply by shmerl** [reply]
Got it, thanks! I'll file a feature request.

**Reply by shmerl** [reply]
Looks like this was added to konsole code very recently:

https://invent.kde.org/utilities/konsole/-/commit/fc772e420a6c64d88376af0ba431850a66d1fc18

Hopefully new release will have this fixed.
